lib/delta/sink/s3.ex

```diff
@@ -34,6 +35,20 @@ defmodule Delta.Sink.S3 do
     end
   end
 
+  defp apply_filename_rewrite(config, filename) do
+    if Map.has_key?(config, :filename_rewrites) do
+      do_apply_filename_rewrite(config.filename_rewrites, filename)
+    else
+      filename
+    end
+  end
+
+  defp do_apply_filename_rewrite(rewrites, filename) do
+    Enum.reduce(rewrites, filename, fn rewrite, modified_filename ->
+      String.replace(modified_filename, rewrite.pattern, rewrite.replacement)
+    end)
+  end
```

Comment:
Maybe a pattern match would be helpful here? It's 6 & 1/2 dozen, so whichever approach you like is fine.
```suggestion
  defp apply_filename_rewrite(%{filename_rewrites: rewrites}, filename) do
    Enum.reduce(rewrites, filename, fn rewrite, modified_filename ->
      String.replace(modified_filename, rewrite.pattern, rewrite.replacement)
    end)
  end
  
  defp apply_filename_rewrite(_config, filename), do: filename
```